server/app/services/applicant/ApplicantService.java

```diff
@@ -1045,4 +1050,118 @@ abstract static class Builder {
+                    new ProgramBlockNotFoundException(programId, blockId));
+              }
+
+              ApplicantQuestion applicantQuestion =
+                  getFirstAddressCorrectionEnabledApplicantQuestion(block.get());
+              AddressQuestion addressQuestion = applicantQuestion.createAddressQuestion();
+
+              Optional<AddressSuggestion> suggestionMaybe =
+                  addressSuggestions.stream()
+                      .filter(
+                          addressSuggestion ->
+                              addressSuggestion.getSingleLineAddress().equals(selectedAddress))
+                      .findFirst();
+
+              ImmutableMap<String, String> questionPathToValueMap =
+                  buildCorrectedAddressAsFormData(
+                      addressQuestion, suggestionMaybe, selectedAddress);
+
+              return CompletableFuture.completedFuture(questionPathToValueMap);
+            });
+  }
+
+  /** Maps address suggestion and corrected state into a form data compatible map */
+  private ImmutableMap<String, String> buildCorrectedAddressAsFormData(
+      AddressQuestion addressQuestion,
+      Optional<AddressSuggestion> suggestionMaybe,
+      String selectedAddress) {
+
+    Map<String, String> questionPathToValueMap = new HashMap<>(Map.of());
+
+    if (suggestionMaybe.isPresent()) {
+      AddressSuggestion suggestion = suggestionMaybe.get();
+      Address address = suggestion.getAddress();
+      AddressLocation location = suggestion.getLocation();
+
+      questionPathToValueMap.put(addressQuestion.getStreetPath().toString(), address.getStreet());
+      questionPathToValueMap.put(addressQuestion.getLine2Path().toString(), address.getLine2());
+      questionPathToValueMap.put(addressQuestion.getCityPath().toString(), address.getCity());
+      questionPathToValueMap.put(addressQuestion.getStatePath().toString(), address.getState());
+      questionPathToValueMap.put(addressQuestion.getZipPath().toString(), address.getZip());
+      questionPathToValueMap.put(
+          addressQuestion.getLatitudePath().toString(), location.getLatitude().toString());
+      questionPathToValueMap.put(
+          addressQuestion.getLongitudePath().toString(), location.getLongitude().toString());
+      questionPathToValueMap.put(
+          addressQuestion.getWellKnownIdPath().toString(), location.getWellKnownId().toString());
+      questionPathToValueMap.put(
+          addressQuestion.getCorrectedPath().toString(),
+          CorrectedAddressState.CORRECTED.toString());
+    } else if (selectedAddress.equals(AddressCorrectionBlockView.USER_KEEPING_ADDRESS_VALUE)) {
+      questionPathToValueMap.put(
+          addressQuestion.getCorrectedPath().toString(),
+          CorrectedAddressState.AS_ENTERED_BY_USER.toString());
+    } else {
+      questionPathToValueMap.put(
+          addressQuestion.getCorrectedPath().toString(), CorrectedAddressState.FAILED.toString());
```

Comment:
**suggestion (blocking):** log a server error here indicating the failure

Reply:
done